### ISSUE TITLE
segfault in GC

### ISSUE THREAD

**Issue by mlubin** [JuliaLang/julia#10600]
```
julia> versioninfo()
Julia Version 0.4.0-dev+3955
Commit 0d8cec3* (2015-03-21 16:58 UTC)
DEBUG build
Platform Info:
  System: Linux (x86_64-linux-gnu)
  CPU: Intel(R) Core(TM) i5-3320M CPU @ 2.60GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Sandybridge)
  LAPACK: libopenblas
  LIBM: libopenlibm
  LLVM: libLLVM-3.3
```
When running ``Pkg.test("JuMP")``, I get the following segfault in the GLPK package.

```
[callback] Test user cuts
     - With solver GLPKMathProgInterface.GLPKInterfaceMIP.GLPKSolverMIP
...
signal (11): Segmentation fault
unknown function (ip: 1127095175)
unknown function (ip: 1127095519)
unknown function (ip: 1127096240)
unknown function (ip: 1127096610)
unknown function (ip: 1127098182)
jl_gc_collect at /mnt/hdd/mlubin/julia-0.4/usr/bin/../lib/libjulia-debug.so (unknown line)
unknown function (ip: 1127090885)
unknown function (ip: 1127091448)
alloc_2w at /mnt/hdd/mlubin/julia-0.4/usr/bin/../lib/libjulia-debug.so (unknown line)
unknown function (ip: 1126946546)
jl_alloc_tuple_uninit at /mnt/hdd/mlubin/julia-0.4/usr/bin/../lib/libjulia-debug.so (unknown line)
jl_alloc_tuple at /mnt/hdd/mlubin/julia-0.4/usr/bin/../lib/libjulia-debug.so (unknown line)
unknown function (ip: 1126444831)
jl_function_ptr at /mnt/hdd/mlubin/julia-0.4/usr/bin/../lib/libjulia-debug.so (unknown line)
_ios_node_is_valid at /home/mlubin/.julia/v0.4/GLPK/src/GLPK_checks.jl:222
ios_node_bound at /home/mlubin/.julia/v0.4/GLPK/src/GLPK.jl:49
_internal_callback at /home/mlubin/.julia/v0.4/GLPKMathProgInterface/src/GLPKInterfaceMIP.jl:164
jlcapi__internal_callback_46216 at  (unknown line)
unknown function (ip: 840348649)
unknown function (ip: 0)
unknown function (ip: 1171693768)
Segmentation fault (core dumped)
```
The offending line is at https://github.com/JuliaOpt/GLPK.jl/blob/f7c0ecfb16c03e755c2c496c13b504e7c2ddc3ec/src/GLPK.jl#L222
If I change it to
```
        err_hook_p = cfunction(_err_hook, Void, (Ptr{Void},)) 
        ccall((:glp_error_hook, libglpk), Void, (Ptr{Void}, Ptr{Void}), err_hook_p, C_NULL)
```
then the segfault appears to go away.

CC @carlobaldassi

**Comment by JeffBezanson** [comment]
@vtjnash Any insight into this?

**Comment by vtjnash** [comment]
no. i think i self-assigned as a reminder to go back and try to reproduce locally

**Comment by vtjnash** [comment]
fixed by afd4eb03 (jl_cfunction_object was missing a null initialization of `sigt`)
